File: Plugins/Aspose_Slides_Java_for_Ruby/lib/asposeslidesjava/Tables/removerowcolumn.rb
module Asposeslidesjava
  module RemoveRowColumn
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/Tables/'
                
        # Create an instance of Presentation class
        pres = Rjb::import('com.aspose.slides.Presentation').new

        # Get the first slide
        slide = pres.getSlides().get_Item(0)

        col_width = [100, 50, 30]
        row_height = [30, 50, 30]                                           
                                                                                       
        table = slide.getShapes().addTable(100, 100, col_width, row_height)
                                                                                       
        table.getRows().removeAt(1, false)                                           
        table.getColumns().removeAt(1, false) 

        # Write the presentation as a PPTX file 
        save_format = Rjb::import('com.aspose.slides.SaveFormat')
        pres.save(data_dir + "RemoveRowColumn.pptx", save_format.Pptx)

        puts "Removed Row & Column from table, please check the output file."
    end
  end
end

File: Examples/src/main/java/com/aspose/slides/examples/Tables/RemoveRowColumn.java
/* 
 * Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
 *
 * This file is part of Aspose.Slides. The source code in this file
 * is only intended as a supplement to the documentation, and is provided
 * "as is", without warranty of any kind, either expressed or implied.
 */
 
package  com.aspose.slides.examples.Tables;

import com.aspose.slides.*;
import com.aspose.slides.examples.Utils;

public class RemoveRowColumn
{
    public static void main(String[] args) throws Exception
    {
        // The path to the documents directory.
        String dataDir = Utils.getDataDir(RemoveRowColumn.class);

        Presentation pres = new Presentation();

        ISlide slide = pres.getSlides().get_Item(0);
        double[] colWidth = { 100, 50, 30 };
        double[] rowHeight = { 30, 50, 30 };

        ITable table = slide.getShapes().addTable(100, 100, colWidth, rowHeight);

        table.getRows().removeAt(1, false);
        table.getColumns().removeAt(1, false);

        pres.save(dataDir+ "TestTable.pptx", SaveFormat.Pptx);
        System.out.println("Program executed successfully");
    }
}




